### PR TITLE
Plugin changes for plumbing k8s events into TaskExecutionEvent

### PR DESCRIPTION
**Pull request by andrewwdye**
# TL;DR
State management and config changes required for plumbing k8s events into `TaskExecutionEvent` as part of https://github.com/flyteorg/flyte/issues/3825

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This change
1. Adds a `TaskInfo.AdditionalReasons` field so that propeller can append k8s event information when handling a k8s phase update.
2. Adds a `SendObjectEvents` option to K8sPluginConfig so this feature can be opted into

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3825

## Follow-up issue
N/A

### CODE REVIEW

File: go/tasks/pluginmachinery/flytek8s/config/config.go [hunk 1]
```diff
@@ -167,6 +168,8 @@ type K8sPluginConfig struct {
 	// DefaultPodTemplateResync defines the frequency at which the k8s informer resyncs the default
 	// pod template resources.
 	DefaultPodTemplateResync config2.Duration `json:"default-pod-template-resync" pflag:",Frequency of resyncing default pod templates"`
+
+	SendObjectEvents bool `json:"send-object-events" pflag:",If true, will send object events as TaskExecutionEvent updates."`
```
Comment (hamersaw) [inline go/tasks/pluginmachinery/flytek8s/config/config.go:172]: Can we add a `k8s` designation here somewhere? `SendK8sObjectEvents` maybe? or in the description. Would be nice to be explicit here.

Comment (andrewwdye) [reply go/tasks/pluginmachinery/flytek8s/config/config.go]: Added to the description since the struct itself already has k8s in the name, but happy to add to the field if you feel that's clearer.

**Review by hamersaw — APPROVED** [review]
